### ISSUE TITLE
If gem depends on git repo, fails with "file exists"

### ISSUE THREAD

**Issue by nateware** [wycats/bundler#148]
[nwiger@sliver:fbapp]$ grep -v '^#' Gemfile |grep -v '^$'
gem "rails", "3.0.pre"
directory "/Users/nwiger/Workspace/rails3", :glob => "{_/,}_.gemspec"
git "git://github.com/rails/arel.git"
git "git://github.com/rails/rack.git"
gem 'mysqlplus', :require_as => 'mysql'
gem 'will_paginate'
gem 'nokogiri'
gem 'facebooker'
gem "aws-s3", :require_as => "aws/s3"
gem 'redis-objects', '>= 0.2.2', :require_as => 'redis/objects'
disable_rubygems

[nwiger@sliver:fbapp]$ gem bundle
Calculating dependencies...
Updating source: http://gems.rubyforge.org
Downloading facebooker-1.0.58.gem
Downloading json-1.2.0.gem
Installing json (1.2.0)
Building native extensions.  This could take a while...
Installing facebooker (1.0.58)
ERROR:  While executing gem ... (Errno::EEXIST)
    File exists - (/Users/nwiger/Workspace/op/proto/fbapp/vendor/gems/dirs/rack, /Users/nwiger/Workspace/op/proto/fbapp/vendor/gems/gems/rack-1.1.0)

**Comment by nateware** [comment]
A little more info, this is triggered by the symlink call (the only one) in /usr/local/lib/ruby/gems/1.9.1/gems/bundler-0.7.2/lib/bundler/bundle.rb:

```
def expand_vendored_gem(spec, options)
  add_spec(spec)
  FileUtils.mkdir_p(@path.join("gems"))
  File.symlink(spec.location, @path.join("gems", spec.full_name))
end
```

Looks like the situation is that the +do_install+ +each+ loop doesn't properly check/remove an existing symlink from gems/gems to dirs

**Comment by jamesarosen** [comment]
I get a similar error when trying to install the dependencies for Rails 3.0.pre on Ruby 1.8.7 (OSX):    $ gem bundle
    Calculating dependencies...
    Fetching git repository at: git://github.com/rails/arel.git
    Updating source: http://gems.rubyforge.org
    Fetching git repository at: git://github.com/rails/arel.git
    ERROR:  While executing gem ... (Errno::EEXIST)
        File exists - /Users/jrosen/projects/ruby/rails/activerecord/railties/../../activesupport or /Users/jrosen/projects/ruby/rails/vendor/gems/ruby/1.8/gems/activesupport-3.0.pre

I've tried uninstalling `activesupport`, but that makes no difference. If it might matter, I did a `gem bundle` from a Rails 3.0.pre app before I tried the one from Rails 3.0.pre itself. (That is I was trying out using Rails before fixing bugs on it.)

**Comment by jamesarosen** [comment]
The problem seems to be that the `railties` directory doesn't exist within `activerecord`. That is, OSX is being picky about that path even though the `../..` backs out of it. Perhaps an `expand_path` is in order?

**Comment by jamesarosen** [comment]
No, that doesn't seem to be it. This particular `gem bundle` seems to expect `activerecord` to include `actionmailer`, `actionpack`, `activemodel`, `activeresource`, and `activesupport` directories, none of which is included in the current Rails 3.0.pre release.
